core/commands/add.go

```diff
@@ -81,6 +81,11 @@ You can now refer to the added file in a gateway, like so:
 			return nil
 		}
 
+		_, found, _ := req.Option(progressOptionName).Bool()
```

Comment:
That seems quite cheaty.
Maybe we should just introduce something like `request agent`? to the `Request` object.